benchmark/python/dot.py

```diff
@@ -17,61 +17,176 @@
 
 import ctypes
 
-from mxnet.test_utils import *
-import scipy.sparse as sp
 import os
 import time
 import argparse
+import subprocess
+import scipy.sparse as sp
 
+from mxnet.test_utils import *
 from mxnet.base import check_call, _LIB
 from util import get_data, estimate_density
 
-parser = argparse.ArgumentParser(description="Benchmark sparse operators",
+PARSER = argparse.ArgumentParser(description="Benchmark sparse operators",
                                  formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-parser.add_argument('--num-omp-threads', type=int, default=1, help='number of omp threads to set in MXNet')
-args = parser.parse_args()
+PARSER.add_argument('--num-omp-threads', type=int,
+                    default=1, help='number of omp threads to set in MXNet')
+PARSER.add_argument('--gpu', action='store_true',
+                    help="to be run on gpu")
+ARGS = PARSER.parse_args()
 
 # some data information
-kdda = {
+KDDA = {
     'data_mini': 'kdda.t.mini',
     'data_name': 'kdda.t',
     'data_origin_name': 'kdda.t.bz2',
     'url': "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary/kdda.t.bz2",
     'feature_dim': 20216830,
-    'm': 200,
-    'batch_size': [64]
+    'm': [1, 8, 32],
+    'batch_size': [64],
+    'default_index': {'batch_size': 0,
+                      'output_dim': 2},
+    'num_batches': 10
 }
 
-avazu = {
+AVAZU = {
     'data_mini': 'avazu-app.t.mini',
     'data_name': 'avazu-app.t',
     'data_origin_name': 'avazu-app.t.bz2',
     'url': "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary/avazu-app.t.bz2",
     'feature_dim': 1000000,
-    'm': 500,
-    'batch_size': [64, 128]
+    'm': [1, 1000, 2000],
+    'batch_size': [128, 256],
+    'default_index': {'batch_size': 0,
+                      'output_dim': 1},
+    'num_batches': 10
 }
 
+CRITEO = {
+    'data_mini': 'criteo.t.mini',
+    'data_name': 'criteo.t',
+    'data_origin_name': 'criteo.t.bz2',
+    'url' : "https://s3-us-west-2.amazonaws.com/sparse-dataset/criteo.t.bz2",
+    'feature_dim': 16000000,
```

Comment:
Could you double check the dimension? It seems to be "2^23 + 13" now, according to Madhav